external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -2602,12 +2589,22 @@ fn parse_dot_or_index_chain(context: &mut Context) -> Result<Exp, Box<Diagnostic
                         }
                     }
                     _ => {
-                        let n = parse_identifier(context)?;
+                        let start_loc = context.tokens.start_loc();
+                        let n = match parse_identifier(context) {
+                            Ok(id) => id,
+                            Err(diag) => {
+                                done_parsing = true;
```

Comment:
Why not just break?

Reply:
If you consider an example I tried to put together in the other [response](https://github.com/MystenLabs/sui/pull/17106#discussion_r1559885274), if we break here (instead of completing construction of `Exp_::Dot`) we will not accomplish our goal as the AST node for the `s.` statement will not be created.

Reply:
Should we just return then?